x-pack/platform/plugins/shared/lens/common/feature_flags.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { FeatureFlagsStart as FeatureFlagsStartPublic } from '@kbn/core/public';
+import type { FeatureFlagsStart as FeatureFlagsStartServer } from '@kbn/core/server';
+
+type GetFlagTypes<T extends Record<string, LensFeatureFlag>> = {
+  [k in keyof T]: T[k]['fallback'];
+};
+
+interface FeatureFlagBase {
+  /**
+   * Unique flag id staring with `lens.` (i.e. `lens.apiFormat`)
+   */
+  id: string;
+}
+
+type BooleanFeatureFlag = FeatureFlagBase & {
+  type: 'boolean';
+  fallback: boolean;
+};
+type NumberFeatureFlag = FeatureFlagBase & {
+  type: 'number';
+  fallback: number;
+};
+type StringFeatureFlag = FeatureFlagBase & {
+  type: 'string';
+  fallback: string;
+};
+
+export type LensFeatureFlag = BooleanFeatureFlag | NumberFeatureFlag | StringFeatureFlag;
+
+export const lensFeatureFlags = {
+  /**
+   * Enables transforming lens state to/from new API Format over the wire.
+   */
+  apiFormat: {
+    id: 'lens.apiFormat',
+    type: 'boolean',
+    fallback: false,
+  },
+} satisfies Record<string, LensFeatureFlag>;
+
+export type LensFeatureFlags = GetFlagTypes<typeof lensFeatureFlags>;
+
+export async function fetchLensFeatureFlags(
+  service: FeatureFlagsStartPublic | FeatureFlagsStartServer
+): Promise<LensFeatureFlags> {
+  const fetchFeatureFlag = fetchFeatureFlagFn(service);
+
+  const flags = await Promise.all(
+    Object.entries(lensFeatureFlags).map(async ([key, flag]) => {
+      const value = await fetchFeatureFlag(flag);
+      return [key, value];
+    })
+  );
+
+  return Object.fromEntries(flags) as LensFeatureFlags;
+}
+
+function fetchFeatureFlagFn(service: FeatureFlagsStartPublic | FeatureFlagsStartServer) {
+  return async function fetchFeatureFlag(flag: LensFeatureFlag) {
```

Comment:
nit: does it need to be `async`?
```suggestion
  return function fetchFeatureFlag(flag: LensFeatureFlag) {
```

Reply:
Sure, updated in 5ca784463fae13a9527eb80e3a14cee2f1c712f1